### PR TITLE
feat: integrate ellipsis

### PR DESCRIPTION
**Pull request by lehainam-dev**
Related bug: https://team-kyber.atlassian.net/browse/AG-452

### CODE REVIEW

File: pkg/source/curve/constant.go [hunk 1]
```diff
@@ -102,3 +102,5 @@ var (
 	emptyString       = ""
 	zero        int64 = 0
 )
+
+var skipInitFactoryDexes = []string{"ellipsis"}
```
Comment (hanguyenk) [inline pkg/source/curve/constant.go:106]: 1. Please add a comment to explain the meaning of this list
2. I think a map would be more efficient in this case

File: pkg/source/curve/embed.go [hunk 2]
```diff
@@ -62,11 +62,18 @@ var optimismPoolsBytes []byte
...
+
 var bytesByPath = map[string][]byte{
 	"pools/arbitrum.json":  arbitrumPoolsBytes,
 	"pools/avalanche.json": avalanchePoolsBytes,
 	"pools/ethereum.json":  ethereumPoolsBytes,
 	"pools/fantom.json":    fantomPoolsBytes,
 	"pools/optimism.json":  optimismPoolsBytes,
 	"pools/polygon.json":   polygonPoolsBytes,
+
+	"pools/ellipsis/bsc.json": ellipsisBscPoolsBytes,
```
Comment (hanguyenk) [inline pkg/source/curve/embed.go:78]: How about `bsc_ellipsis.json`?

File: pkg/source/curve/pools_list_updater.go [hunk 3]
```diff
@@ -65,37 +67,39 @@ func (d *PoolsListUpdater) GetNewPools(ctx context.Context, metadataBytes []byte
...
-		newPoolLimitLeft = newPoolLimitLeft - (nextOffset - registryOrFactoryList[i].Offset)
-
-		for j := 0; j < len(poolAddresses); j++ {
-			poolTypeMap[poolTypes[j]] = append(poolTypeMap[poolTypes[j]], PoolAndRegistries{
-				PoolAddress:              poolAddresses[j],
-				RegistryOrFactoryABI:     &registryOrFactoryList[i].ABI,
-				RegistryOrFactoryAddress: &registryOrFactoryList[i].Address,
-			})
-		}
+	if !skipInitFactory(d.config.DexID) {
```
Comment (hanguyenk) [inline pkg/source/curve/pools_list_updater.go:70]: please use early return here

**Review by it4rb — APPROVED** [review]

File: pkg/source/curve/embed.go (hunk 2 above)
Comment (lehainam-dev) [reply pkg/source/curve/embed.go:78]: I follow the same convention with `saddle`, where each dex is grouped into a single folder under the `pools` folder.
I think I still keep `pools/ellipsis/bsc.json` to be consistent with others.

File: pkg/source/curve/constant.go (hunk 1 above)
Comment (lehainam-dev) [reply pkg/source/curve/constant.go:106]: Fixed in the latest commit.

File: pkg/source/curve/pools_list_updater.go (hunk 3 above)
Comment (lehainam-dev) [reply pkg/source/curve/pools_list_updater.go:70]: Below the condition, there's a shared logic so because of that, I can't use early return here.

**Review by hanguyenk — APPROVED** [review]

**Pull request merged**